deps/eventlog/src/ccel/device_paths/bios_boot_spec_parser.rs

```diff
@@ -0,0 +1,88 @@
+use super::{DevicePathParser, DeviceSubTypeParser};
+use anyhow::Result;
+use scroll::{Pread, LE};
+use std::collections::HashMap;
+use std::sync::LazyLock;
+
+/// Parser for Type 5 - BIOS Boot Specification Device Path.
+/// Define in section 10.3.6 of <https://uefi.org/specs/UEFI/2.10/10_Protocols_Device_Path_Protocol.html#bios-boot-specification-device-path>
+static BIOS_SUBTYPE_HANDLERS: LazyLock<HashMap<u8, Box<dyn DeviceSubTypeParser + Send + Sync>>> =
+    LazyLock::new(|| {
+        let mut handlers: HashMap<u8, Box<dyn DeviceSubTypeParser + Send + Sync>> = HashMap::new();
+        handlers.insert(0x01, Box::new(BbsParser));
+        // Add more subtype parsers here as needed
+        handlers
+    });
+
+pub struct BiosBootSpecParser;
+
+impl DevicePathParser for BiosBootSpecParser {
+    fn parse(&self, sub_type: u8, data: &[u8]) -> Result<String> {
+        BIOS_SUBTYPE_HANDLERS
+            .get(&sub_type)
+            .ok_or_else(|| {
+                anyhow::anyhow!("Unknown Bios Boot Specification subtype: {:#04x}", sub_type)
+            })?
+            .parse(data)
+    }
+}
+
+pub struct BbsParser;
+
+impl DeviceSubTypeParser for BbsParser {
+    fn parse(&self, data: &[u8]) -> Result<String> {
+        let mut index = 0;
+
+        let device_type = match data
+            .gread_with::<u16>(&mut index, LE)
+            .map_err(|e| anyhow::anyhow!("Failed to read device type: {:?}", e))?
+        {
+            0x01 => "FLOPPY",
+            0x02 => "HARD_DRIVE",
+            0x03 => "CDROM",
+            0x04 => "PCMCIA",
+            0x05 => "USB",
+            0x06 => "EMBEDDED_NETWORK",
+            0x80 => "DEV",
+            0xFF => "UNKNOWN",
+            other => &*other.to_string(),
+        };
+
+        let status_flag: u16 = data
+            .gread_with(&mut index, LE)
+            .map_err(|e| anyhow::anyhow!("Failed to read status flag: {:?}", e))?;
+
+        let desc_bytes = &data[index..];
+
+        let desc_end = desc_bytes
+            .iter()
+            .position(|&b| b == 0)
+            .unwrap_or(desc_bytes.len());
+
+        let description = std::str::from_utf8(&desc_bytes[..desc_end])
+            .map_err(|e| anyhow::anyhow!("Failed to parse description: {:?}", e))?
+            .to_string();
```

Comment:
is this String transformation required? using &str should be ok in the format! macro, no?

Reply:
Yeah - thanks -> fixed